annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastProcessor.java

```diff
@@ -9,88 +10,226 @@
 import javax.annotation.processing.*;
 import javax.lang.model.SourceVersion;
 import javax.lang.model.element.Element;
+import javax.lang.model.element.PackageElement;
 import javax.lang.model.element.TypeElement;
 import javax.tools.Diagnostic;
+import javax.tools.FileObject;
 import javax.tools.JavaFileObject;
-import java.io.IOException;
-import java.io.PrintWriter;
+import javax.tools.StandardLocation;
+import java.io.*;
+import java.nio.file.Paths;
 import java.util.*;
+import java.util.stream.Collectors;
 
-@SupportedAnnotationTypes("com.exactpro.epfast.annotations.FastType")
+@SupportedAnnotationTypes({"com.exactpro.epfast.annotations.FastType", "com.exactpro.epfast.annotations.FastPackage"})
 @SupportedSourceVersion(SourceVersion.RELEASE_8)
 @AutoService(Processor.class)
 public class FastProcessor extends AbstractProcessor {
+    private static final TypeName SERVICE_TYPENAME = new TypeName("com.exactpro.epfast.ICreator");
+
     private MustacheFactory mustacheFactory = new DefaultMustacheFactory();
 
     private TypeElement fastTypeElement;
 
+    private TypeElement fastPackageElement;
+
+    private HashMap<PackageElement, HashMap<String, Element>> packageElementsMap;
+
     @Override
     public synchronized void init(ProcessingEnvironment processingEnv) {
         super.init(processingEnv);
         fastTypeElement = processingEnv.getElementUtils().getTypeElement(FastType.class.getCanonicalName());
+        fastPackageElement = processingEnv.getElementUtils().getTypeElement(FastPackage.class.getCanonicalName());
```

Comment:
If you call some method several times and this method supposed to return the same or interchangeable value it's better to store the result of method call in variable. You never know if library method logic is simple or complex and resource-consuming